src/sentry/snuba/query_subscription_consumer.py

```diff
@@ -280,9 +282,23 @@ def handle_message(self, message: Message) -> None:
                     },
                 )
                 try:
+                    # XXX(ahmed): Temporary hack to be able to extract entity key from query
+                    # which is now required by snuba because with the introduction of metrics
+                    # dataset, the relationship between dataset and entity is no longer 1-to-1
+                    # However, will deploy a fix in snuba to send the entity key in the payload
+                    # of the message
+                    entity_regex = r"(MATCH|match)[ ]*\(([^)]+)\)"
```

Comment:
Would anchor the regex at the beginning with `^`. Otherwise it may _may_ match within a string literal, maybe?